### DISCUSSION TITLE
Stress free temperature

### DISCUSSION THREAD

**Discussion by deepakbiitk** [idaholab/moose#24783] [Q&A General] [answered]
Hi all,

In  "ComputeThermalExpansionEigenstrain" Can we provide a function in stress free temperature? In most examples I saw, it takes an constant value?

Thanks,
Deepak

**Comment by GiudGiud** [comment]
Hello

You ll want to check the other strain models for functions.
You want a functions right? like a pre-determined function of space and time, not a variable

Guillaume

**Reply by deepakbiitk** [reply]
Yes, I would like to input a function varying in space as stress_free_temperature instead of a constant value.

Deepak

**Comment by hugary1995** [comment, marked as answer]
The stress free temperature takes a variable. 

> I would like to input a function varying in space as stress_free_temperature instead of a constant value.

You can define a spatially varying auxvariable using your function.

**Reply by GiudGiud** [reply]
Use a FunctionAux for that
https://mooseframework.inl.gov/source/auxkernels/FunctionAux.html
